File: src/test/java/io/binac/leetcode/ScrambleStringTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class ScrambleStringTest {
    private final ScrambleString.Solution1 solution1 = new ScrambleString.Solution1();

    @Test
    void test1() {
        assertTrue(solution1.isScramble("great", "rgeat"));
        assertFalse(solution1.isScramble("abcde", "caebd"));
    }
}

File: cpp/unittests/ScrambleStringTest.cpp
#include "ScrambleString.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(ScrambleString, Solution87_1) {
  EXPECT_TRUE(Solution87_1().isScramble("great", "rgeat"));
  EXPECT_FALSE(Solution87_1().isScramble("abcde", "caebd"));
}

TEST(ScrambleString, Solution87_2) {
  EXPECT_TRUE(Solution87_2().isScramble("great", "rgeat"));
  EXPECT_FALSE(Solution87_2().isScramble("abcde", "caebd"));
}
